rust/bin/agent/src/main.rs

```diff
@@ -367,6 +368,31 @@ impl algorithm::ANN for QBGService {
     }
 }
 
+fn parse_duration_from_string(input: &str) -> Option<Duration> {
+    if input.len() < 2 {
+        return None;
+    }
+    let last_char = match input.chars().last() {
+        Some(c) => c,
+        None => return None,
+    };
+    if last_char.is_numeric() {
+        return None;
+    }
+
+    let (value, unit) = input.split_at(input.len() - 1);
+    let num: u64 = match value.parse() {
+        Ok(n) => n,
+        Err(_) => return None,
+    };
+    match unit {
+        "s" => Some(Duration::from_secs(num)),
```

Comment:
[nits] Do we need errors like "Format like 11m20s is not supported, please use 680s" ?